### PR TITLE
fix(learn): Change demo urls on various backend projects

### PR DESCRIPTION
**Pull request by RandellDawson**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

This PR updates the demo projects to use the ones hosted by freeCodeCamp for the 17 Node back-end projects and 3 Node proxies.  It also updates the regular expression for certain tests which validate the demo project url is not submitted.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/take-home-projects/show-the-local-weather.english.md
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/take-home-projects/show-the-local-weather.english.md:16]:
```suggestion
<strong>Note:</strong> Many internet browsers now require an HTTP Secure (<code>https://</code>) connection to obtain a user's locale via HTML5 Geolocation. For this reason, we recommend using HTML5 Geolocation to get user location and then use the freeCodeCamp Weather API <a href='https://weather-proxy.freecodecamp.rocks/' target='_blank'>https://weather-proxy.freecodecamp.rocks/</a> which uses an HTTP Secure connection for the weather. Also, be sure to connect to <a href='https://codepen.io' target='_blank'>CodePen.io</a> via <code>https://</code>.
```

**Review by naomi-lgbt — COMMENTED** [review]
A small typo in one of the links.

File: curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.english.md
Comment (Sky020) [inline curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.english.md:29]:
```suggestion
        assert(!/.*\/stock-price-checker-proxy\.freecodecamp\.rocks/.test(getUserInput('url')));
```

Comment (nhcarrigan) [reply curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.english.md:29]: Wait I thought the `proxy` url was for the API, not the example project.

Comment (Sky020) [reply curriculum/challenges/english/09-information-security/information-security-projects/stock-price-checker.english.md:29]: @nhcarrigan Yes, it is. That is why I made this suggestion...to immediately remove it 🤦‍♂️

**Comment by moT01** [comment]
I wonder if we should include both the replit and .rocks sites in the tests for `I can provide my own project, not the example URL.` I don't really mind either way.

**Comment by nhcarrigan** [comment]
Will the repl.it examples continue to remain live?

**Comment by RandellDawson** [comment]
> Will the repl.it examples continue to remain live?

They will, but there will no longer be any reference to them from learn.  I could add both, but I don not think it will be necessary.  The Glitch one is out there too, but just like the repl.it projects, it would be difficult to find them.

File: curriculum/challenges/english/06-quality-assurance/quality-assurance-projects/metric-imperial-converter.md
Comment (moT01) [inline curriculum/challenges/english/06-quality-assurance/quality-assurance-projects/metric-imperial-converter.md:10]:
```suggestion
Build a full stack JavaScript app that is functionally similar to this: <a href="https://metric-imperial-converter.freecodecamp.rocks/" target="_blank">https://metric-imperial-converter.freecodecamp.rocks/</a>.
```

File: curriculum/challenges/english/10-coding-interview-prep/take-home-projects/show-the-local-weather.md
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/take-home-projects/show-the-local-weather.md:16]:
```suggestion
<strong>Note:</strong> Many internet browsers now require an HTTP Secure (<code>https://</code>) connection to obtain a user's locale via HTML5 Geolocation. For this reason, we recommend using HTML5 Geolocation to get user location and then use the freeCodeCamp Weather API <a href='https://weather-proxy.freecodecamp.rocks/' target='_blank'>https://weather-proxy.freecodecamp.rocks/</a> which uses an HTTP Secure connection for the weather. Also, be sure to connect to <a href='https://codepen.io' target='_blank'>CodePen.io</a> via <code>https://</code>.
```

**Review by moT01 — CHANGES_REQUESTED** [review]
Couple typos

**Review by moT01 — APPROVED** [review]
LGTM 🎉 

Any final thoughts on that server? Roll with it for now - maybe make some 404 pages later?

**Comment by RandellDawson** [comment]
> Any final thoughts on that server? Roll with it for now - maybe make some 404 pages later?

@moT01 Let's wait for @raisedadead to comment on that.

**Comment by raisedadead** [comment]
Thanks for your patience guys, while I hash out the pages and wind up on the config (which I have not been able to come up with a good way) please go ahead and make a PR to the `production-current` branch. 

I can land that to unblock you? As I understand the only things that are pending are the actual go live?

**Comment by moT01** [comment]
To be clear @raisedadead, are you saying to move this PR to target `production-current`. We aren't exactly blocked, I feel like if we are still working on that server, we should probably get it where we want it before landing this so the projects don't stop working if we need to make adjustments there. Let me know if you want a second pair of eyes on that.

**Comment by raisedadead** [comment]
We will not be able to land this on production separately. 

Randell confirmed that it is going to take more than just updating URLs on production. The new hosting server is ready and stable from an Ops standpoint. It does lack some "nice to haves", which I will circle around when we have more time.

I am leaving this PR blocked as is, and should merge it when we are ready to go live with the current master.

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
So we got on a call and discussed pushing master out the door, blocking #39799 instead.

**Pull request merged**
